### PR TITLE
DQMGUI Render plugin for use with Integration tests

### PR DESCRIPTION
**Pull request by OrangGeeGee**
Introducing a dedicated Render plugin in DQMGUI to be used in Continuous Integration.
Changes are discussed with @rovere @vanbesien

### CODE REVIEW

File: dqmgui/style/IntegrationTestRenderPlugin.cc [hunk 1]
```diff
@@ -0,0 +1,258 @@
+/*!
+  \file SiPixelRenderPlugin
+  \brief Display Plugin for Pixel DQM Histograms
+  \author P.Merkel
+  \version $Revision: 1.49 $
+  \date $Date: 2012/04/10 09:44:09 $
+*/
```
Comment (vanbesien) [inline dqmgui/style/IntegrationTestRenderPlugin.cc]: This needs an update.
Would be nice to state here why this renderplugin was created and what we intend to do with it.

**Comment by OrangGeeGee** [comment]
@vanbesien good catch :) Updated the comment.

**Comment by vanbesien** [comment]
Ok from my side. Unless @rovere has anything to add, we can request Lina and Bruno to merge it.

File: dqmgui/style/IntegrationTestRenderPlugin.cc [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+
+    name_text = new TText(0.05,0.01, o.name.c_str());
+    name_text->SetTextColor(kBlack);
+    name_text->SetTextSize(0.02);
+    name_text->SetNDC();
+    name_text->Draw("same");
+  }
+
+  // simple recursive descent parser for the "Column(0,30,50,)/Other(0,3,5,)/Last"
+  // format used to carry the positions where histograms where concatenated by
```
Comment (rovere) [inline dqmgui/style/IntegrationTestRenderPlugin.cc]: typo in comment

File: dqmgui/style/IntegrationTestRenderPlugin.cc [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+  // EXTEND to this plugin.
+  template<typename Iterator>
+  struct LabelMarkerParser {
+    std::vector<std::vector<int>> markers; // output
+    std::string text; // cleaned-up label
+
+    static std::pair<bool, int> parse_int(Iterator& start, Iterator end) {
```
Comment (rovere) [inline dqmgui/style/IntegrationTestRenderPlugin.cc]: Any reason to have ref and by-value here?
I see (value, value) later on. Puzzled

File: dqmgui/style/IntegrationTestRenderPlugin.cc [hunk 4]
```diff
@@ -0,0 +1,262 @@
...
+      int out = 0;
+      if (start == end || !std::isdigit(*start)) return std::make_pair(false, out);
+      while (start != end && std::isdigit(*start))
+        out = out * 10 + (*start++ - '0');
```
Comment (rovere) [inline dqmgui/style/IntegrationTestRenderPlugin.cc]: Maybe that's the reason...

**Comment by rovere** [comment]
I'm not sure I fully follow the convoluted logic of markers and label handling, but maybe that's because some logic is embedded in the title axis of the histogram that is later on decoded: can I see an example of a plot rendered by this plugin?

**Comment by OrangGeeGee** [comment]
Attached the example results.
The convoluted logic is why I picked SiPixelRenderPlugin as a base.

integrationtestth1
integrationtestth2
overlay

**Comment by OrangGeeGee** [comment]
@rovere We still need to merge this! Can you ping those who can do that?

**Comment by threus** [comment]
@BrunoCoimbra @h4d4 could you please merge this PR?
Thanks,
Tomas for DQM

**Pull request merged**